### DISCUSSION TITLE
Version 5.4.0.0

### DISCUSSION THREAD

**Discussion by rsjaffe** [rsjaffe/MIDI2LR#661] [Announcements]
## New features
  * New masking commands: Invert group, Duplicate and invert group.
  * Local correction (mask) amount.
  * Preset amount (when preset amount slider is enabled).
  * New languages: हिन्दी, norsk (Bokmål) and polski.
  * Note for हिन्दी:- इंस्टॉलर हिंदी का समर्थन नहीं करता है। हालांकि, आवेदन करता है। MIDI2LR इंस्टॉलर के लिए उपयोग की जाने वाली भाषा की परवाह किए बिना सभी भाषाओं का समर्थन करता है।
## Improvements
  * Installer now has a native ARM version (included in app bundle for MacOS).
  * Reduced load on Lightroom when refreshing MIDI controller data.
  * Better shutdown behavior. May reduce shutdown hangs that are occurring with MacOS ARM processors.
  * Adobe has improved its documentation of the API. As a result, some reverse-engineered items in MIDI2LR that used to be marked "experimental" are no longer experimental. See [list of commands](https://github.com/rsjaffe/MIDI2LR/wiki/Commands) for details.
  * **CropAngle** command has been deprecated for quite some time and doesn't work as reliably as **straightenAngle**. Now, all uses of CropAngle in a profile will silently be corrected to straightenAngle and CropAngle is removed from the command list.
## Bug fixes
  * Windows installer now works on Windows 11.

<hr /><em>This discussion was created from the release <a href='https://github.com/rsjaffe/MIDI2LR/releases/tag/v5.4.0.0'>Version 5.4.0.0</a>.</em>

**Comment by felipeboca** [comment]
Hi! Update file (dmg) is not opening on my MBP m1 2020. Should I do something different to update the app?

**Reply by rsjaffe** [reply]
I've reuploaded the file. Could you download it again and try again?

**Reply by felipeboca** [reply]
The DMG file opens a disk image and inside it the file MIDI2LR-5.4.0.0-osx-installer.app, but clicking on the item to open does not load any installation window.  🥹

**Reply by rsjaffe** [reply]
OK, I'll delete it, rebuild the application and installer, and reupload it later today.

**Reply by rsjaffe** [reply]
I rebuilt the installer, and ran it on my Mac (x64 computer) and it ran fine. It's now uploaded in the release.

**Reply by ilquare** [reply]
I tried now on mac air 2020 M1 and the installation does not start

**Reply by Creek211** [reply]
I am having the same experience on my M1 Studio. Redownloaded a couple of minutes ago and same result. After I confirm opening, nothing.

**Reply by rsjaffe** [reply]
OK, I've rebuilt it without an ARM-specific installer. Sounds like there's an issue with Installbuilder support of ARM installers. Could you redownload it and report back on whether it works? Thanks.

**Reply by Creek211** [reply]
It now works fine on my M1 Studio; downloaded 15 June 8:58AM PST. Thanks!

**Comment by Enricopez** [comment]
The new Version 5.4.0.0 have bring a lot of problem during editing. Every 2 seconds an error pop-up the error is this one : [https://drive.google.com/file/d/1kcF1jXrj-AlapiR8Ouosw0L9uTvTvqRi/view?usp=sharing](url)
Someone can point me on the old version? is more stable.

**Reply by rsjaffe** [reply]
The link is not working, so I can't see the error. The releases are at https://github.com/rsjaffe/MIDI2LR/releases. Is this the first time you've used MIDI2LR?

**Reply by Enricopez** [reply]
this is the error that continues to cause problems : https://www.dropbox.com/s/0uajxcjxkl6hkiw/Schermata%202022-06-15%20alle%2023.28.10.png?dl=0
@rsjaffe I've been using MIDI2LR for years and I follow you every time you release a new release. ☺️
I edited 800 images between last night and this morning and with the old release it never gave any problems. This evening I went to the mac and I installed the new release and it immediately gave me problems. There is something wrong 🤷🏻‍♂️

**Reply by rsjaffe** [reply]
Could you post the MIDI2LR.log file?  This file is found by using the Lightroom menus: Help → Plug-in Extras MIDI2LR → Generate diagnostic report.

**Reply by Enricopez** [reply]
Hi @rsjaffe , thank for your help... here the file:
[MIDI2LR.log](https://github.com/rsjaffe/MIDI2LR/files/8916671/MIDI2LR.log)

At the moment I had to reinstall the previous version because the last one kept giving me an error and slowed down my editing.
If I need to reinstall it to generate another log let me know.
🙏🏻 thanks

**Reply by ilquare** [reply]
Always slow to close lightroom ... the old problem has not been solved :-(

**Reply by ilquare** [reply]
rsjaffe: Do you not have a mac with m1 to try?

**Reply by rsjaffe** [reply]
Nope. Don't have an m1. This is free open source software, and I would appreciate it if someone with an m1 would debug the issue. I cannot. I could also use help from someone with ASIO expertise, which is where I think the problem arises.

**Reply by rsjaffe** [reply]
> Hi @rsjaffe , thank for your help... here the file: [MIDI2LR.log](https://github.com/rsjaffe/MIDI2LR/files/8916671/MIDI2LR.log)
> 
> At the moment I had to reinstall the previous version because the last one kept giving me an error and slowed down my editing. If I need to reinstall it to generate another log let me know.

I can't see the issue there. Can you attach the profile file you're using when you get the errors?

**Reply by dtwarogpl** [reply]
Hello, exactly same problem here. Im using windows, last lightroom classic version installed (11.4)

**Reply by dtwarogpl** [reply]
@rsjaffe here is my log file. 
[MIDI2LR.log](https://github.com/rsjaffe/MIDI2LR/files/8943213/MIDI2LR.log)

**Reply by rsjaffe** [reply]
The commonality between the two reports is that the system language and Lightroom language are different. 

One has a system language of Polish and a Polish (Programmers) keyboard and is using lightroom in the English language. For this one, I also see that sending keystrokes fails with "Wystąpił błąd potwierdzenia." (google translate says "A confirmation error has occurred.")

The other has a system language of Italian and a Lightroom language of English. Keyboard type is Italian-Pro, It looks like keystrokes were sent without problem.

I'm unable to do any debugging for a few months, and frankly, this one will be hard to figure out. My suggestion in the meantime is to try Lightroom in the same language as the system language, and see if the problem goes away.

**Comment by cyrkielfotograf** [comment]
@rsjaffe i can't instal app on Ventura OS :( like i don't have permissions 
 
 https://ibb.co/tZdqXfd

**Reply by rsjaffe** [reply]
Can you run the installer with elevated permissions? Ventura's not coming out until September, I believe. Other than giving you a few suggestions as to what to try I can't help you. I suspect the installer vendor (Installbuilder) will have to come out with an update.

**Reply by rsjaffe** [reply]
Czy możesz uruchomić instalator z podwyższonymi uprawnieniami? Ventura wyjdzie dopiero we wrześniu, jak sądzę. Poza kilkoma sugestiami dotyczącymi tego, czego spróbować, nie mogę ci pomóc. Podejrzewam, że dostawca instalatora (Installbuilder) będzie musiał wydać aktualizację.

**Comment by cyrkielfotograf** [comment]
I make downgrade to Monterey and everything just work perfect.

**Comment by cstsang** [comment]
I'm getting the same "an internal error has occurred: ?:0: attempt to perform arithmetic on field '?" (a nil value) everytime I go into the masking tool (M) and midi2lr tries to load the file for "masking."
[MIDI2LR.log](https://github.com/rsjaffe/MIDI2LR/files/8962310/MIDI2LR.log)

There is no error if I leave the filename out of that box and leave it blank.  I get the same error if I put anything in the box, regardless if it is a valid file, or missing file name.

**Reply by rsjaffe** [reply]
Thanks for narrowing down the issue. I am unable to fix this until September.

**Reply by cstsang** [reply]
No problem, I'll roll back one version until them.  thanks!

**Reply by kernelo-panico** [reply]
Exactly the same issue here, thanks for looking into this! Otherwise phenomenal plugin, keep up the tremendous work you're doing!

**Comment by ka249** [comment]
tell me how to run the setup please. there is no exe

**Reply by ka249** [reply]
the problem is solved)

**Comment by jrendant** [comment]
I have a new computer running windows 11. Lightroom version 12.0 and the latest Midi2LR.  I create a directory for the profiles to be saved into and then try to save a file to it. While the save dialog box opens and I can enter a filename, nothing gets saved. Tried pressing buttons and rotating dials but nothing populates in the program. I am most likely missing something here and I need some help, please.

**Reply by jrendant** [reply]
I solved the issue. I ran Midi2LR externally and then launched LR I had 2 instances of Midi2LR and I was using the wrong one although it did say it was connected to LR.
